app/uk/gov/hmrc/exports/services/notifications/NotificationFactory.scala

```diff
@@ -17,38 +17,28 @@
 package uk.gov.hmrc.exports.services.notifications
 
 import javax.inject.Inject
-import play.api.Logger
+import play.api.Logging
 import uk.gov.hmrc.exports.models.declaration.notifications.Notification
 
 import scala.util.{Failure, Success, Try}
-import scala.xml.{NodeSeq, XML}
+import scala.xml.XML
 
-class NotificationFactory @Inject()(notificationParser: NotificationParser) {
-
-  private val logger = Logger(this.getClass)
+class NotificationFactory @Inject()(notificationParser: NotificationParser) extends Logging {
 
   def buildNotifications(actionId: String, notificationXml: String): Seq[Notification] =
     Try(XML.loadString(notificationXml)).map(notificationParser.parse) match {
```

Comment:
`notificationParser.parse` can throw an exception if the xml is valid but does not contain the required elements. In this scenario `logParseExceptionAtPagerDutyLevel` will not be called (and I think it should).

Reply:
There is a case for `Failure(exc)` which is executed when either `XML.loadString` or `notificationParser.parse` throws any exception.
Also, there are tests in `NotificationFactory` (near the end of the file) that cover the scenario when NotificationParser throws an exception. Although it is not tested if logger is called, empty sequence is returned in case of a `Failure`. And there is a test that asserts against return value.